src/core/types/keys/diffie_hellman/private_key.rs

```diff
@@ -0,0 +1,195 @@
+use crate::prelude::*;
+use crypto::keys::x25519::SecretKey as X25519PrivateKey;
+use hkdf::Hkdf;
+use k256::sha2::Sha256;
+
+#[derive(derive_more::Debug)]
+#[debug("{}", self.to_hex())]
+pub struct DiffieHellmanPrivateKey(X25519PrivateKey);
+
+impl DiffieHellmanPrivateKey {
+    pub fn generate() -> Result<Self> {
+        Exactly32Bytes::generate().to_vec().try_into()
+    }
+
+    pub fn public_key(&self) -> DiffieHellmanPublicKey {
+        self.0.public_key().into()
+    }
+
+    pub fn hkdf_key_agreement(
+        &self,
+        other: &DiffieHellmanPublicKey,
+        salt: &[u8],
```

Comment:
can be:
`salt: impl AsRef<[u8]>` and use with `as_ref()` in body below, that way we can pass `BagOfBytes`, `Vec`, or `ExactlyNBytes` types as salt 

and do same for `into` ?

Reply:
maybe time for us to do a `pub type SymmetricKey = Exactly32Bytes` typealias? and use it here and in VersionedEncryption (and its impl, i.e. search for `encryption_key` and `decryption_key`), I think it is more clear.